multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -72,6 +73,7 @@ func NewStaleResCleanupController(
 	reconciler := &StaleResCleanupController{
 		Client:           Client,
 		Scheme:           Scheme,
+		ctx:              context.Background(),
```

Comment:
Making a background context a member variable is perhaps meaningless. The context can be used to gracefully shutdown the controller, you can:
1. create a context from the stopCh via `wait.ContextForChannel(stopCh)` in `Run`.
2. pass the context to methods that need it.

This can achieve that once the stopCh is closed, the controller will stop what it's doing and exit immediately.

Reply:
Done, thanks.

Reply:
But now the code is exposed to race condition. And it's actually more common and perhaps preferred to pass context via method parameters, instead of being stored as a member variable.
See https://pkg.go.dev/context
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:
>
>  ```
> func DoSomething(ctx context.Context, arg Arg) error {
>	// ... use ctx ...
> }
> ```

Reply:
Got it, I will refine this. Thanks.